mmv1/third_party/terraform/services/tpuv2/resource_tpu_v2_vm_test.go.erb

```diff
@@ -35,63 +30,139 @@ func TestAccTpuV2Vm_update(t *testing.T) {
+			},
+			{
+				ResourceName:            "google_tpu_v2_vm.tpu",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"labels", "terraform_labels", "zone"},
 			},
 			{
-				Config: testAccTpuV2Vm_update(updatedContext),
+				Config: testAccTpuV2Vm_update(context, false),
 			},
 			{
 				ResourceName:            "google_tpu_v2_vm.tpu",
 				ImportState:             true,
 				ImportStateVerify:       true,
-				ImportStateVerifyIgnore: []string{"zone"},
+				ImportStateVerifyIgnore: []string{"labels", "terraform_labels", "zone"},
 			},
 		},
 	})
 }
 
 func testAccTpuV2Vm_full(context map[string]interface{}) string {
 	return acctest.Nprintf(`
-data "google_tpu_v2_runtime_versions" "available" {
+resource "google_tpu_v2_vm" "tpu" {
   provider = google-beta
+
+  name        = "tf-test-test-tpu-%{random_suffix}"
+  zone        = "us-central1-c"
+  description = "Text description of the TPU."
+
+  runtime_version  = "tpu-vm-tf-2.13.0"
+  accelerator_type = "v2-8"
+
+  scheduling_config {
+    preemptible = true
+  }
+
+  data_disks {
+    source_disk = google_compute_disk.disk.id
+    mode        = "READ_ONLY"
+  }
+
+  labels = {
+    foo = "bar"
+  }
+
+  metadata = {
+    foo = "bar"
+  }
+
+  tags = ["foo"]
+
+  lifecycle {
+    prevent_destroy = true
+  }
 }
 
-data "google_tpu_v2_accelerator_types" "available" {
+resource "google_compute_disk" "disk" {
   provider = google-beta
+
+  name  = "tf-test-tpu-%{random_suffix}"
+  image = "debian-cloud/debian-11"
+  size  = 10
+  type  = "pd-ssd"
+  zone  = "us-central1-c"
 }
+`, context)
+}
+
+func testAccTpuV2Vm_update(context map[string]interface{}, preventDestroy bool) string {
+	context["prevent_destroy"] = strconv.FormatBool(preventDestroy)
 
+	return acctest.Nprintf(`
 resource "google_tpu_v2_vm" "tpu" {
   provider = google-beta
 
-  name = "tf-test-test-tpu%{random_suffix}"
-  zone = "us-central1-c"
-  description = "%{description}"
+  name        = "tf-test-test-tpu-%{random_suffix}"
+  zone        = "us-central1-c"
+  description = "Text description of the TPU updated."
 
   runtime_version  = "tpu-vm-tf-2.13.0"
   accelerator_type = "v2-8"
-}
-`, context)
-}
 
-func testAccTpuV2Vm_update(context map[string]interface{}) string {
-	return acctest.Nprintf(`
-data "google_tpu_v2_runtime_versions" "available" {
-  provider = google-beta
+  scheduling_config {
+    preemptible = true
+  }
+
+  data_disks {
+    source_disk = google_compute_disk.disk.id
+    mode        = "READ_WRITE"
+  }
+
+  data_disks {
+    source_disk = google_compute_disk.disk2.id
+    mode        = "READ_ONLY"
+  }
+
+  labels = {
+    baz = "bar"
+  }
+
+  metadata = {
+    baz = "bar"
+  }
+
+  tags = ["baz"]
+
+  lifecycle {
+    prevent_destroy = %{prevent_destroy}
+  }
 }
 
-data "google_tpu_v2_accelerator_types" "available" {
+resource "google_compute_disk" "disk" {
   provider = google-beta
+
+  name  = "tf-test-tpu-%{random_suffix}"
```

Comment:
nit: can we give these disks better names? I was thinking it might be misleading when debugging tests if it's called "tpu" rather than something with "disk"

Reply:
This choice is largely from the perspective of our team seeing resources accumulating in our nightly test projects, and that may not match typical usage. But basically, when a bunch of disks are left in the instance, it is typically easier to resolve if we can identify which test the resource came from (ie. a tpu test).

Happy to change this to align to other usage though. Maybe `tf-test-tpu-disk-` is fine?

Reply:
`tf-test-tpu-disk-` sounds like a good compromise to me